### PR TITLE
Add functionalities for creating posts, comments, and likes

### PR DESCRIPTION
**Pull request by PabloBona**
### Proposed Changes:

- **Implemented a `current_user` Method:**
- Created a `current_user` method in the ApplicationController, providing universal access to the current user's data across all controllers. Currently, it retrieves the initial user from the database without authentication, a feature scheduled for implementation in forthcoming updates.

**Created Forms and Enabled Functionalities:**

 **Post Creation:**
- Users can now create posts by clicking the "New Post" button. They will need to add a title and text to the post. 
- Implemented a user-friendly form enabling users to craft posts with unique titles and content.

**Comment Creation:**
- Users can now create comments by clicking the "Add Comment" button. They will need to add a text to the comment. 
- Designed an intuitive form allowing users to compose comments tailored to specific posts.

**Adding Likes to Posts:**
  - Users can express their appreciation by adding "likes" to posts, enhancing interactivity and engagement within the application.
  - Integrated a functionality allowing users to seamlessly like posts, contributing to a more interactive user experience.

**Enhanced Seed Data:**

- **Diverse User Profiles:**
  - Implemented a comprehensive seed script that populates the database with diverse users, each accompanied by a unique profile photo and biography.
  - Ensured a rich and varied content landscape by seeding posts attributed to individual users, promoting a dynamic user experience.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -6,6 +6,30 @@ def index
...
+  def create
+    @user = User.find(params[:user_id])
+    @post = @user.posts.build(post_params)
+
+    if @post.save
+      redirect_to user_post_path(@user, @post), notice: 'Post created successfully.'
+    else
+      render :new
+    end
+  end
```
Comment (levy002) [inline app/controllers/posts_controller.rb]: - [ ] Kindly make sure a new post is created on behalf of the currently logged-in user (current_user) of the app in order to match the project requirements.

```suggestion
  def create
    @user = current_user
    @post = @user.posts.build(post_params)

    if @post.save
      redirect_to user_post_path(@user, @post), notice: 'Post created successfully.'
    else
      render :new
    end
  end
```

**Review by levy002 — CHANGES_REQUESTED** [review]
## Hello @PabloBona   , :wave:
### I'm Levy, your PR reviewer

Good job so far :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- No linter error :heavy_check_mark: 

## Status: REQUIRED CHANGES ♻️
_Check the comments under the review._


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @levy002 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (PabloBona) [reply app/controllers/posts_controller.rb]: @levy002 Thanks for your feedback! I have already made the changes and am requesting another review.

Comment (levy002) [reply app/controllers/posts_controller.rb]: Hello @PabloBona , thanks for implementing requested changes. Kindly go ahead and submit your PR link in your dashboard, you will receive the feedback in no time. Happy Friday ☺️

**Review by anagudelogu — APPROVED** [review]
# STATUS: APPROVED! 🍾 🎆 🥇 

Hi @PabloBona, 
I'm Andres.

You did a great job working on this milestone, your project is complete! 🚀 
There is nothing else to say other than... it's time to merge it 🤖  
Congratulations! 🎉 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**